lib/DependencyScan/ModuleDependencyScanner.cpp

```diff
@@ -742,13 +742,13 @@ ModuleDependencyScanner::resolveAllClangModuleDependencies(
   // results back to the shared set for future lookups.
   const llvm::DenseSet<clang::tooling::dependencies::ModuleID> seenClangModules =
       cache.getAlreadySeenClangModules();
-  std::mutex cacheAccessLock;
+  std::mutex lookupResultLock;
```

Comment:
I think this guards both `cache` and `moduleLookupResult` now? Not sure which name is better. I might prefer two fine grain lock inside both data structure to be easy to use.

Also since you expect `moduleLookupResult` to grow anyway, do you need to populated with `std::nullopt`?

Reply:
In this section of the code, it was just a rename to be consistent, but I will leave it as-is for now to focus this PR on the actual functional fix that's required. 

The `nullopt` may not necessarily be needed, but is currently used to indicate that no lookup was actually performed due to a cache hit.